tests/Unit/DataStructures/Tensor/Test_TensorData.py

```diff
@@ -0,0 +1,62 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre.DataStructures import DataVector
+from spectre.DataStructures import ExtentsAndTensorVolumeData
+from spectre.DataStructures import TensorComponent
+import unittest
+import numpy as np
+import numpy.testing as npt
+
+
+class TestTensorData(unittest.TestCase):
+    # Tests for TensorComponent functions
+    def test_tensor_component(self):
+        # Test name
+        tc1 = TensorComponent("tensor component", DataVector([1.5, 1.1]))
+        self.assertEqual(tc1.name, "tensor component")
+        tc1.name = "new tensor component"
+        self.assertEqual(tc1.name, "new tensor component")
+
+        # Test data
+        tc2 = TensorComponent("tensor component", DataVector([1.5, 1.1]))
```

Comment:
One reason why I suggested to combine these test cases was so you don't have to set up the thing multiple times. It's a very simple class, no need to write such a long test. This is sufficient:

```py
tc = TensorComponent("tensor component", DataVector([1.5, 1.1]))
self.assertEqual(tc.name, "tensor component")
self.assertEqual(tc.data, DataVector([1.5, 1.1]))
self.assertEqual(str(tc), "(tensor component, (1.5,1.1))")
self.assertEqual(repr(tc), "(tensor component, (1.5,1.1))")
tc.name = "new tensor component"
self.assertEqual(tc.name, "new tensor component")
tc.data = DataVector([1., 2.])
self.assertEqual(tc.data, DataVector([1., 2.]))
```

(similar for the `ExtentsAndTensorVolumeData` test)